oauth/middleware.py

```diff
@@ -0,0 +1,26 @@
+from django.contrib.auth import authenticate
+from django.http import HttpResponse
+
+
+class OAuth2TokenMiddleware:
+    """
+    Django 1.10 has changed the middlewares signature.
+    oauth2_provider.middleware.OAuth2TokenMiddleware is still old style.
+    https://github.com/evonove/django-oauth-toolkit/issues/406
```

Comment:
Fix committed: https://github.com/evonove/django-oauth-toolkit/commit/911f7b828ec70ca02aab111a660412bd1e90a7c2 (but not yet released).